src/utils/wallet.js

```diff
@@ -191,7 +315,16 @@ class Wallet {
     }
 
     async removeAccessKey(publicKey) {
-        return await this.getAccount(this.accountId).deleteKey(publicKey)
+        const { account, has2fa } = await this.getAccountAndState()
+        if (has2fa) {
+            const request = {
+                receiver_id: this.accountId,
+                actions: [{ type: 'DeleteKey', public_key: splitPK(publicKey) }]
+            }
+            return await this.makeTwoFactorRequest(account, request)
+        } else {
+            return await this.getAccount(this.accountId).deleteKey(publicKey)
+        }
     }
 
     async removeNonLedgerAccessKeys() {
```

Comment:
not sure why there are changes to this method

Reply:
to handle the 2fa case